### ISSUE TITLE
Add more spellchecking info about locale and dictionary to INSTALL.md

### ISSUE THREAD

**Issue by yurivict** [qTox/qTox#5603]
qTox is built with ```-DSPELL_CHECK=ON```. Spell checking doesn't seem to work. Typing misspelled words doesn't cause them to be labeled as incorrect.

The ```INSTALL``` file only says that the above option needs to be turned on. But does it also need to be enabled in some way? ```Spell checking``` checkbox is checked.

OS: FreeBSD 11.2
qTox version: 1.16.3
Commit hash:  bef9d4b7
toxcore:  0.2.9
Qt:  5.12.1
kf5-sonnet-5.56.0

**Comment by sudden6** [comment]
what does the log output say? Do you have any dictionaries installed?

**Comment by yurivict** [comment]
The log contains this message ~50 times:
```
[15:18:53.873 UTC] :0 : Warning: No language dictionaries for the language: "C"
```

The INSTALL file https://github.com/qTox/qTox/blob/master/INSTALL.md only says that ```-DSPELL_CHECK=ON``` disables spell checking, and ```sonnet>= 5.45``` is required.

This INSTALL file is probably missing information on what else is required in order for spell checking to work. I've never heard of sonnet before, hearing about it for the first time. The INSTALL file should assume that users know nothing about the required libraries, and should mention all necessary steps.

**Comment by anthonybilinski** [comment]
Your locale must be set through the environment variable `$LANG`.
Your locale should be updated by editing `/etc/locale.gen` then running `# locale-gen`: https://wiki.archlinux.org/index.php/locale#Generating_locales

or setting it in your DE: https://wiki.archlinux.org/index.php/locale#My_system_is_still_using_wrong_language

**Comment by yurivict** [comment]
So the INSTALL file should say that locale is expected.
Now it says:
```
[17:04:34.802 UTC] :0 : Warning: No language dictionaries for the language: "en_US"
```

**Comment by anthonybilinski** [comment]
You now must install a dictionary for any languages you want supported, e.g.
`hunspell-en-us` (on Ubuntu). Sonnet supports multiple dictionary backends - not just hunspell.

I'll look at updating `INSTALL.md`, I think locale should normally be set for users but we can have a footnote in case things aren't working, and I think mentioning that dictionaries need to be installed is reasonable.

**Comment by yurivict** [comment]
I think a one-time popup error message in case spellchecking can't work is also reasonable.
It should explain why it can't work: "Wrong locale", "No dictionaries are installed", etc.

**Comment by yurivict** [comment]
Many devs don't bother to set the locale, so the locale ends up to be "C" (?) In such case it should be assumed that the language is English, and qTox spell checking should default to English. This would make spell checking much easier to use for many.

**Comment by devasia2112** [comment]
ehllo, i'm trying to figure out how to use spellchecking with qTox, I have sonnet 5.59.0 installed. 

> $`hunspell -D` tells me I have:
`/usr/share/hunspell/en_US`
`/usr/share/hunspell/en_US-large`
`/usr/share/myspell/dicts/en_US`
`/usr/share/myspell/dicts/en_US-large`

> $`echo $LANG` tells me:
`en_US.UTF-8`

I'm able to use `hunspell` with another program, but it's not working with `qTox`, and my log have no mentions of dictionaries, languages or anything like that.

Regards,

**Comment by Diadlo** [comment]
@deepcell Did you build qTox with `-DSPELL_CHECK=ON`?

**Comment by devasia2112** [comment]
@Diadlo no, I installed via pacman on ArchLinux. I will remove qTox and compile from the source then. Thanks a lot.

**Comment by yurivict** [comment]
I have qTox built as a port with ```-DSPELL_CHECK=ON```, installed on FreeBSD. ```hunspell``` is installed. I never saw spell check working, not even once.

What is wrong?

**Comment by devasia2112** [comment]
Is there any way to check if qTox has `-DSPELL_CHECK=ON` or  `DSPELL_CHECK=OFF` ? Also I did read somewhere in the issues that  `DSPELL_CHECK` is by default ON. Thanks.

**Comment by yurivict** [comment]
@deepcell You can grep for SPELL_CHECK, the build laves the value in the cmake cache file.

**Comment by devasia2112** [comment]
@yurivict all right thx.

**Comment by Diadlo** [comment]
@yurivict If you understand C++, you can try to build `sonnet` from source with debug info and try to debug it. Unfortunately, I don't see another way now.

Also, you can try to build the newest version of sonnet from source (looks like it's 5.60.0) and try it

Try to build examples from sonnet repo. Do it works for you?

**Comment by yurivict** [comment]
> Also, you can try to build the newest version of sonnet from source (looks like it's 5.60.0) and try it

I have ```kf5-sonnet-5.57.0``` installed.

**Comment by yurivict** [comment]
For the record: I reported the bug to KDE that ```sonnet``` fails to perform spell checking.
https://bugs.kde.org/show_bug.cgi?id=408923

**Comment by Diadlo** [comment]
> I have `kf5-sonnet-5.57.0` installed.

Yes, it's why I suggest building the newest version (probably not released yet)
